packages/openapi-ts/src/plugins/@hey-api/client-next/bundle/client.ts

```diff
@@ -57,12 +57,15 @@ export const createClient = (config: Config = {}): Client => {
       await opts.requestValidator(opts);
     }
 
-    if (opts.body && opts.bodySerializer) {
+    if (opts.body !== undefined && opts.bodySerializer) {
       opts.serializedBody = opts.bodySerializer(opts.body);
     }
 
     // remove Content-Type header if body is empty to avoid sending invalid requests
-    if (opts.serializedBody === undefined || opts.serializedBody === '') {
+    if (
+      opts.body === undefined ||
+      (opts.serializedBody !== undefined && opts.serializedBody === '')
```

Comment:
The first check is redundant

```suggestion
      (opts.serializedBody === '')
```

Reply:
good point - thanks, I will update in both clients.